### PR TITLE
Oracle: Set the right Tx version

### PR DESCRIPTION
**Pull request by shargon**
- Assign the right version when we create a new TX
- The type of error in the response was also eliminated, we should log it, but not hash or retransmit it to reduce indeterminism

### CODE REVIEW

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 1]
```diff
@@ -40,6 +40,7 @@ public class Transaction : IEquatable<Transaction>, IInventory, IInteroperable
         private Cosigner[] cosigners;
         private byte[] script;
         private Witness[] witnesses;
+        private UInt256 oracleRequestTx;
```
Comment (Tommo-L) [inline src/neo/Network/P2P/Payloads/Transaction.cs:43]: We create a new field! Why not use attribute?

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 2]
```diff
@@ -121,6 +128,11 @@ public int Size
                         Cosigners.GetVarSize() +    //Cosigners
                         Script.GetVarSize() +       //Script
                         Witnesses.GetVarSize();     //Witnesses
+
+                    if (Version == TransactionVersion.OracleResponse)
+                    {
```
Comment (Tommo-L) [inline src/neo/Network/P2P/Payloads/Transaction.cs:133]: `TransactionType` will be better? It seems we are misusing version and type?  @roman-khimov I think this [comment](https://github.com/neo-project/neo/issues/1274#issuecomment-605117812) makes sense.

File: src/neo/Network/P2P/Payloads/Transaction.cs (hunk 1 above)
Comment (shargon) [reply src/neo/Network/P2P/Payloads/Transaction.cs:43]: https://github.com/neo-project/neo/pull/1539#issuecomment-609380165

File: src/neo/Network/P2P/Payloads/Transaction.cs (hunk 2 above)
Comment (roman-khimov) [reply src/neo/Network/P2P/Payloads/Transaction.cs:133]: True, types are types and versions are versions, both have their own valid meaning and purpose and it's better not to mix them.

Comment (shargon) [reply src/neo/Network/P2P/Payloads/Transaction.cs:133]: The original idea was this https://github.com/neo-project/neo/pull/1539#issuecomment-609378672

File: src/neo/Network/P2P/Payloads/Transaction.cs (hunk 1 above)
Comment (roman-khimov) [reply src/neo/Network/P2P/Payloads/Transaction.cs:43]: It's interesting to me to compare this implementation to attribute-based one, for some reason I think that reusing attributes would make things easier and require less changes. And adding this field to each and every transaction is a bit inefficient memory-wise.

**Comment by shargon** [comment]
I think that we could change the way later, there are no agrement between how to check that it's a oracle TX, and where put the `DependOn` information, so i will add this in my TODO list and I will merge this PR, it coud be improved or change it later, before merge to `master`

- Option A: VersionType and extend the properties (the current one)
- Option B: Using only Attributes.

**Pull request merged**